website/src/views/home.js

```diff
@@ -23,13 +24,24 @@ function Home() {
     const binsInfo = useFetch('/bins');
     const gradeInfo = useFetch(`students/${fetchEmail}/grades`);
     const projectionsInfo = useFetch(`/students/${fetchEmail}/projections`);
+    const navigate = useNavigate();
 
     useEffect(() => {
         if (binsInfo.data && localStorage.getItem('token')) {
             setBinsData(binsInfo.data.map(({ letter, points }) => [points, letter]));
         }
     }, [binsInfo.data]);
 
+
+    // Redirect to error page if any of the fetch requests return an error
+    useEffect(() => {
+        if (binsInfo.error || gradeInfo.error || projectionsInfo.error) {
+            const errorCode = binsInfo.error?.status || gradeInfo.error?.status || projectionsInfo.error?.status;
+            const errorText = binsInfo.error?.message || gradeInfo.error?.message || projectionsInfo.error?.message;
```

Comment:
Be careful here.  Only assign these values if they match the case.  eg: if `binsInfo.error= = { status: 500 }` and `gradeInfo.error == { status: 404, message: "not found" }`, then you will be directed to a 500 error page with the message "not found".